model/riscv_insts_base.sail

```diff
@@ -295,10 +295,20 @@ mapping clause assembly = RTYPE(rs2, rs1, rd, op)
 /* ****************************************************************** */
 union clause ast = LOAD : (bits(12), regidx, regidx, bool, word_width, bool, bool)
 
+/* Loads only need the a 2 bit size encoding as quads are not supported.
+The 3rd bit is replaced by a sign bit for loads. */
+
+mapping strip : bits(2) <-> bits(3) = {
```

Comment:
I think rather than this, which seems like a bit of a hack, it would make more sense to just add a 2-bit version of `size_enc` (i.e. the original `size_enc`). Like `size_enc_narrow` or something.

Reply:
Don't we then get two definitions of `BYTE`, `HALF`, `WORD`  and `DOUBLE`? It would be good to have a single source of truth on width.

Reply:
Maybe if we did something like signed_size_enc that takes in the word_width and sign bit and returns a 3 bit encoding with the sign bit as the most significant bit and the 2 bit encoding in the lower 2 bits. That might be cleaner than just stripping off a bit and avoids the ambiguity of size_enc and size_enc_narrow.

Reply:
In essence, that what the `strip` mapping does implicitly. But I agree, we could make that more explicit.

Reply:
@martinberger @Timmmm So would we rather do a 2 bit mapping to a narrower `size_enc` or an additional 3 bit mapping that matches based on both sign and `word_width`?

Reply:
I mean something like this:

```
mapping size_enc : word_width <-> bits(2) = {
  BYTE   <-> 0b00,
  HALF   <-> 0b01,
  WORD   <-> 0b10,
  DOUBLE <-> 0b11,
}

mapping size_enc_wide : word_width <-> bits(3) = {
  BYTE   <-> 0b000,
  HALF   <-> 0b001,
  WORD   <-> 0b010,
  DOUBLE <-> 0b011,
  QUAD <-> 0b100,
}
```

I originally suggested `size_enc_narrow` / `size_enc` but if you do `size_enc` / `size_enc_wide` then there's no need to change any of the existing uses. Though that may be desirable anyway because it looks like the `0b0 @ size_enc(..)` is really intended to one day be `size_enc_wide(..)`.

> Don't we then get two definitions of BYTE, HALF, WORD and DOUBLE? 

Two definitions of the encodings... kind of. I don't think it really matters in this case. They're not going to change and there are only 4. But I guess if you really wanted to avoid that you could do:

```
mapping size_enc_wide : word_width <-> bits(3) = {
  QUAD <-> 0b100,
  narrow <-> 0b0 @ size_enc(narrow),
}
```

Reply:
That's a good solution, too!

Reply:
I went with size_enc_wide referencing size_enc for consistency and clarity about what it is doing.